scripts/ci/pre_commit/pre_commit_check_eager_constraints.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+set -euo pipefail
+# shellcheck source=scripts/ci/libraries/_script_init.sh
+. "$( dirname "${BASH_SOURCE[0]}" )/../libraries/_script_init.sh"
+
+
+set_eager_upgrade_additional_requirements_variable
+
+PROD_DOCKER_FILE_EAGER_CONSTRAINTS=$(grep "ARG EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS=" Dockerfile | cut -d'"' -f 2)
+CI_DOCKER_FILE_EAGER_CONSTRAINTS=$(grep "ARG EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS=" Dockerfile.ci | cut -d'"' -f 2)
+
+error="false"
+if [[ ${PROD_DOCKER_FILE_EAGER_CONSTRAINTS} != "${EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS_VARIABLE}" ]]; then
+    echo
+    echo """
```

Comment:
```suggestion
    echo -e """
```

Reply:
We do not need it here. Our variables are already defined in the way that does not need backlash interpretation:

```
COLOR_RESET=$'\e[0m'
```

I am also planning to do in for the in_container variables as I noticed it's not done there this way.